### PR TITLE
MGMT-1113: Linting commands - including Prettier

### PR DESCRIPTION
**Pull request by celdrake**
- Fixed the issues with the dependencies that impeded running the linter

- Converted the files to TS files. Most code has been left untouched, as there were many errors to fix at once.

### CODE REVIEW

**Review by jgyselov — DISMISSED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -5,21 +5,24 @@
   "overrides": [
     {
       "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
-      "parserOptions": {
-        "project": ["packages/*/tsconfig(.*)?.json"]
-      },
```
Comment (jkilzi) [inline .eslintrc.json]: Why did you remove this?
This allows @typescript-eslint to lint files using information from the type-checker

File: packages/integration-tests/.eslintrc.json [hunk 2]
```diff
@@ -1,10 +1,21 @@
 {
   "extends": ["plugin:cypress/recommended", "../../.eslintrc.json"],
-  "ignorePatterns": ["!**/*"],
+  "ignorePatterns": ["!**/*", "node_modules/*"],
```
Comment (jkilzi) [inline packages/integration-tests/.eslintrc.json:3]: [node_modules and .dotfiles are always, implicitly, ignored](https://eslint.org/docs/latest/user-guide/configuring/ignoring-code#the-eslintignore-file)
```suggestion
  "ignorePatterns": ["!**/*"],
```

File: packages/integration-tests/.eslintrc.json [hunk 3]
```diff
@@ -1,10 +1,21 @@
...
   "overrides": [
     {
       "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
-      "rules": {}
+      "parserOptions": {
+        "project": ["packages/integration-tests/tsconfig.json"]
+      },
```
Comment (jkilzi) [inline packages/integration-tests/.eslintrc.json]: We had this same part at the top level...

File: packages/integration-tests/src/Cypress.d.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+interface AIInterceptsConfig {
+  activeSignal: string;
+  activeScenario: string;
+}
```
Comment (jkilzi) [inline packages/integration-tests/src/Cypress.d.ts]: It may be better to move it to its own .d.ts file not to pollute the global type namespace.

File: packages/integration-tests/src/fixtures/hosts/index.ts [hunk 5]
```diff
@@ -29,17 +27,15 @@ const getUpdatedHosts = () => {
     return [];
   }
 
-  const lastSignal = Cypress.env('AI_LAST_SIGNAL');
+  const lastSignal: string = Cypress.env('AI_LAST_SIGNAL');
```
Comment (jkilzi) [inline packages/integration-tests/src/fixtures/hosts/index.ts:30]: Is  `const lastSignal: string` always a string? What if `AI_LAST_SIGNAL` is undefined?

File: packages/integration-tests/src/fixtures/infra-envs/index.ts [hunk 6]
```diff
@@ -1,14 +1,12 @@
...
 import { fakeClusterId, fakeClusterInfraEnvId } from '../cluster/base-cluster';
 
 const infraEnv = {
   id: fakeClusterInfraEnvId,
   cluster_id: fakeClusterId,
   href: `/api/assisted-install/v2/infra-envs/${fakeClusterInfraEnvId}`,
-  download_url: `${Cypress.env(
-    'API_BASE_URL',
-  )}/images/${fakeClusterInfraEnvId}?arch=x86_64&type=full-iso&version=4.9`,
+  download_url: `${Cypress.env('API_BASE_URL')}/images/${fakeClusterInfraEnvId}?arch=x86_64&type=full-iso&version=4.9`,
```
Comment (jkilzi) [inline packages/integration-tests/src/fixtures/infra-envs/index.ts]:
```suggestion
  download_url: `${Cypress.env('API_BASE_URL')}/images/${fakeClusterInfraEnvId}?arch=x86_64&type=full-iso&version=4.9`,
```

Wouldn't be better to have a function that generates these URLs?

**Review by jkilzi — CHANGES_REQUESTED** [review]

File: packages/integration-tests/.eslintrc.json (hunk 2 above)
Comment (celdrake) [reply packages/integration-tests/.eslintrc.json:3]: Then it must be something having to do with `nx`, because linting with the above configuration takes about 8 secons, while if not excluding `node_modules`, I've waited longer than 1 minute before cancelling the process.

So I'd keep it.

File: .eslintrc.json (hunk 1 above)
Comment (celdrake) [reply .eslintrc.json]: I'll revert this and remove the one I added in the `integration-tests` project

File: packages/integration-tests/src/fixtures/hosts/index.ts (hunk 5 above)
Comment (celdrake) [reply packages/integration-tests/src/fixtures/hosts/index.ts:30]: Since we asserted at the beginning that we are in an scenario where at least 1 host is discovered, it would always have a value.

File: packages/integration-tests/src/fixtures/infra-envs/index.ts (hunk 6 above)
Comment (celdrake) [reply packages/integration-tests/src/fixtures/infra-envs/index.ts]: It's just a constant, but it could be reused in a few places so I did that

File: packages/integration-tests/.eslintrc.json [hunk 7]
```diff
@@ -1,10 +1,18 @@
 {
   "extends": ["plugin:cypress/recommended", "../../.eslintrc.json"],
-  "ignorePatterns": ["!**/*"],
+  "ignorePatterns": ["!**/*", "node_modules/*"],
```
Comment (jkilzi) [inline packages/integration-tests/.eslintrc.json:3]: [node_modules and .dotfiles are always ignored](https://eslint.org/docs/latest/user-guide/configuring/ignoring-code#:~:text=In%20addition%20to%20any%20patterns%20in%20the%20.eslintignore%20file%2C%20ESLint%20always%20follows%20a%20couple%20of%20implicit%20ignore%20rules%20even%20if%20the%20%2D%2Dno%2Dignore%20flag%20is%20passed.)

File: packages/integration-tests/package.json [hunk 8]
```diff
@@ -4,48 +4,26 @@
...
     "lodash": "^4.17.15",
-    "prettier": "^2.0.1",
     "uuid": "^8.1.0",
     "yaml": "^1.7.2"
   },
-  "scripts": {
-    "lint": "TODO",
-    "lint:fix": "TODO"
-  },
+  "scripts": {},
```
Comment (jkilzi) [inline packages/integration-tests/package.json:18]:
```suggestion
```

**Review by jkilzi — APPROVED** [review]

File: packages/integration-tests/.eslintrc.json (hunk 2 above)
Comment (jkilzi) [reply packages/integration-tests/.eslintrc.json:3]: well `"!**/*"` it is telling to not ignore anything... why is it there?

Comment (celdrake) [reply packages/integration-tests/.eslintrc.json:3]: Looks like it may have to do with the configuration they use in the "lint" action. If removed, it fails
lint-patterns
